test/hotspot/jtreg/serviceability/jvmti/vthread/VThreadEventTest/VThreadEventTest.java

```diff
@@ -0,0 +1,210 @@
+ * The reason is we want the state of tested thread to be predictable.
+ * With original CountDownLatch it is not clear what thread state is expected.
+ */
+class CountDownLatch {
+    private int count = 0;
+
+    CountDownLatch(int count) {
+        this.count = count;
+    }
+
+    public synchronized void countDown() {
+        count--;
+        notify();
+    }
+
+    public synchronized void await() throws InterruptedException {
+        while (count > 0) {
+            wait(1);
+        }
+    }
+}
+
+public class VThreadEventTest {
+    static final int TCNT1 = 10;
+    static final int TCNT2 = 4;
+    static final int TCNT3 = 4;
+    static final int THREAD_CNT = TCNT1 + TCNT2 + TCNT3;
+    static final long TIMEOUT_BASE = 1_000_000L;
+
+    private static void log(String msg) { System.out.println(msg); }
+
+    private static native int threadEndCount();
+    private static native int threadMountCount();
+    private static native int threadUnmountCount();
+
+    private static volatile boolean attached;
+    private static boolean failed;
+    private static List<Thread> test1Threads = new ArrayList(TCNT1);
+
+    private static CountDownLatch ready0 = new CountDownLatch(THREAD_CNT);
+    private static CountDownLatch ready1 = new CountDownLatch(TCNT1);
+    private static CountDownLatch ready2 = new CountDownLatch(THREAD_CNT);
+    private static CountDownLatch mready = new CountDownLatch(1);
+
+    private static void await(CountDownLatch dumpedLatch) {
+        try {
+            dumpedLatch.await();
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    // The test1 vthreads are kept unmounted until interrupted after agent attach.
+    static final Runnable test1 = () -> {
+        synchronized (test1Threads) {
+            test1Threads.add(Thread.currentThread());
+        }
+        log("test1 vthread started");
+        ready0.countDown();
+        await(mready);
+        ready1.countDown(); // to guaranty state is not State.WAITING after await(mready)
+        try {
+            Thread.sleep(20000); // big timeout to keep unmounted untill interrupted
+        } catch (InterruptedException ex) {
+            // it is expected, ignore
+        }
+        ready2.countDown();
+    };
+
+    // The test2 vthreads are kept mounted until agent attach.
+    static final Runnable test2 = () -> {
+        log("test2 vthread started");
+        ready0.countDown();
+        await(mready);
+        while (!attached) {
+            // keep mounted
+        }
+        ready2.countDown();
+    };
+
+    // The test3 vthreads are kept mounted until agent attach.
+    static final Runnable test3 = () -> {
+        log("test3 vthread started");
+        ready0.countDown();
+        await(mready);
+        while (!attached) {
+            // keep mounted
+        }
+        LockSupport.parkNanos(10 * TIMEOUT_BASE); // will cause extra mount and unmount
```

Comment:
I don't see much sense in TIMEOUT_BASE constant (it's used only here and multiplied by 10)
I think it would be clearer to
```suggestion
        // park for 10ms; causes extra unmount and mount
        LockSupport.parkNanos(10_000_000L);
```

Reply:
Thanks - fixed.